app/src/androidTest/java/ch/epfl/sdp/appart/LoginUITest.java

```diff
@@ -30,12 +35,44 @@
 import static org.hamcrest.Matchers.allOf;
 
 @LargeTest
-@RunWith(AndroidJUnit4.class)
+//@RunWith(AndroidJUnit4.class)
+@UninstallModules(FireBaseModule.class)
+@HiltAndroidTest
 public class LoginUITest {
 
-    @Rule
+    @Rule(order = 0)
+    public HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+    @Rule(order = 1)
     public ActivityTestRule<MainActivity> mActivityTestRule = new ActivityTestRule<>(MainActivity.class);
 
+    @BindValue
+    Database database = new MockDataBase();
```

Comment:
Is this where you specify which Database implementation you want ?

Reply:
In the tests yes, but you can also use mockito to overwrite the behaviors of some methods.